x/ibc/20-transfer/module.go

```diff
@@ -238,15 +239,71 @@ func (am AppModule) OnRecvPacket(
 	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
 	}
-	return handlePacketDataTransfer(ctx, am.keeper, packet, data)
+	acknowledgement := FungibleTokenPacketAcknowledgement{
+		Success: true,
+		Error:   "",
+	}
+	if err := am.keeper.OnRecvPacket(ctx, packet, data); err != nil {
+		acknowledgement = FungibleTokenPacketAcknowledgement{
```

Comment:
Currently any error returned by `keeper.OnRecvPacket` will cause a failed acknowledgement to be committed rather than reverting the transaction. Is this the behavior we want?

Reply:
Yes, that is the behaviour we want. Reverting the transaction would mean that the sequence number (e.g. in an ordered channel) would never proceed, which we don't want - IBC packets always have to be executed.